modules/diskq/qdisk.c

```diff
@@ -437,7 +437,8 @@ qdisk_pop_head(QDisk *self, GString *record)
         {
           self->hdr->backlog_head = self->hdr->read_head;
 
-          if (!self->options->read_only && self->hdr->length == 0)
+          g_assert(self->hdr->backlog_len == 0);
```

Comment:
@gaborznagy I think this deserves a separate commit with some explanation.
What do you think?

Reply:
Discussed IRL, this belongs to the patch.